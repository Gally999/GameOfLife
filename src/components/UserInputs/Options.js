import React from 'react';
import PropTypes from "prop-types";

import GridInputs from './GridInputs';

const Options = ({ x, y, onDimensionsChange }) => (
  <div>
    <h4>Please choose the dimensions of your grid:</h4>
    <GridInputs
      x={x}
      y={y}
      onDimensionsChange={onDimensionsChange}
    />
  </div>
);

Options.propTypes = {
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
  onDimensionsChange: PropTypes.func.isRequired,
}

export default Options;
