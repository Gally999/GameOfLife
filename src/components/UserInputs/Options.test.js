import React from 'react';
import Options from './Options';

describe('Options', () => {
  it('renders the options component correctly', () => {
    const subject = (
      <Options
        x={4}
        y={7}
        onDimensionsChange={() => {}}
      />
    );
    expect(subject).toMatchSnapshot();
  });
});
