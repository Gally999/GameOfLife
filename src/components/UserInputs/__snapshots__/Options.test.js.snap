// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options renders the options component correctly 1`] = `
<Options
  onDimensionsChange={[Function]}
  x={4}
  y={7}
/>
`;
